GVFS/GVFS.Common/PlaceholderListDatabase.cs

```diff
@@ -218,18 +278,29 @@ private bool TryParseRemoveLine(string line, out string key, out string error)
 
         public class PlaceholderData
         {
-            public PlaceholderData(string path, string sha)
+            public PlaceholderData(string path, string fileShaOrFolderValue)
             {
                 this.Path = path;
-                this.Sha = sha;
+                this.Sha = fileShaOrFolderValue;
             }
 
             public string Path { get; }
             public string Sha { get; }
 
             public bool IsFolder
             {
-                get { return this.Sha == GVFSConstants.AllZeroSha; }
+                get 
+                {
+                    return this.Sha.StartsWith(FolderValuePrefix, StringComparison.Ordinal);

```

Comment:
Why not just compare the full strings? Are the prefixes creating a significant perf win?